app/components/user-status.hbs

```diff
@@ -0,0 +1,24 @@
+<div class="heading">
+  {{#if (eq @status "active")}}
+    <h2 data-test-status>I am doing a task</h2>
+  {{/if}}
+  {{#if (eq @status "idle")}}
+    <h2 data-test-status>I am idle</h2>
+  {{/if}}
+  {{#if (eq @status "ooo")}}
+    <h2 data-test-status>I am OOO</h2>
+  {{/if}}
+</div>
+<div class="buttons">
+  {{#if (eq @status "active")}}
+    <button class="buttons__idle" type="button" {{on "click" (fn @onUpdateStatus 'idle')}}><span data-test-button-text>Change status to ‘Idle’</span></button>

```

Comment:
Seems like a lot of repeating code, can we simplify this using an object?

Reply:
Yess, I am working on the same.

Reply:
For the first two cases (active and idle) it can be simplied but problem is with 3rd one (ooo) it is a bit different from the other two. I have simplified for the headings but for buttons, unable to find any different way.
```
<div class="heading">
  {{#if @status}}
    <h2 data-test-status>I am {{@status}}</h2>
  {{/if}}
</div>

<div class="buttons">
  {{#if (eq @status "active")}}
    <button class="buttons__idle" type="button" {{on "click" (fn @onUpdateStatus 'idle')}}><span data-test-button-text>Change status to ‘Idle’</span></button>
    <button class="buttons--small" type="button" {{on "click" (fn @onUpdateStatus 'ooo' )}}><span>Mark yourself as OOO</span></button>
  {{/if}}
  {{#if (eq @status "idle")}}
    <button class="buttons__active" type="button" {{on "click" (fn @onUpdateStatus 'active')}}><span data-test-button-text>Change status to ‘Active’</span></button>
    <button class="buttons--small" type="button" {{on "click" (fn @onUpdateStatus 'ooo' )}}><span>Mark yourself as OOO</span></button>
  {{/if}}
  {{#if (eq @status "ooo")}}
    <button class="buttons--small" type="button" {{on "click" (fn @onUpdateStatus 'active' )}}><span>Mark yourself as Active again</span></button>
  {{/if}}
</div>
```

Reply:
The logic for heading doesn't sound right to me, the text which is to be shown to the user is different from the task status itself

Reply:
Yess it is only different for `active` one. For rest two statuses it is same as in design. I think we can add a js file to manage these strings but how to implement it I am not getting, there is way like we can use something like this,
```
@tracked currentUserStatus = 'I am Idle';
  @action update() {
    this.currentUserStatus = 'I am doing a task';
  }
  ```
But here I am not sure how to get returned data here from the route and set strings accordingly. Still finding a way to implement this.